azalea-protocol/src/packets/game/clientbound_initialize_border_packet.rs

```diff
@@ -16,3 +16,19 @@ pub struct ClientboundInitializeBorderPacket {
     #[var]
     pub warning_time: u32,
 }
+
+impl McBufWritable for ClientboundInitializeBorderPacket {
+    fn write_into(&self, buf: &mut impl std::io::Write) -> Result<(), std::io::Error> {
+        self.new_center_x.write_into(buf)?;
+        self.new_center_z.write_into(buf)?;
+        self.old_size.write_into(buf)?;
+        self.new_size.write_into(buf)?;
+        self.lerp_time.var_write_into(buf)?;
+        // Odd
```

Comment:
This is very suspicious. Maybe the VarLong implementation is wrong?

Reply:
Oh, it is a bug with VarLong. It's supposed to write a zero into the buffer at the end.
Here's the function from PacketByteBuf:

public PacketByteBuf writeVarLong(long value) {
    while((value & -128L) != 0L) {
        this.writeByte((int)(value & 127L) | 128);
            value >>>= 7;
    }

    this.writeByte((int)value);
    return this;
}